routes/admin-login.js

```diff
@@ -1,8 +1,54 @@
 var express = require('express');
 var router = express.Router();
+var firebase = require('firebase');
+var config = require('./config.js');
+
+if(!firebase.apps.length)
+	firebase.initializeApp(config.config);
 
 router.get('/', function(req, res, next) {
 	res.render('admin-login');
 });
 
+router.post('/', signIn, function(req, res) {
+	//checks if user is signed in
+	if(firebase.auth().currentUser)
+		res.jsonp({success: true});
+
+	//sign in unsuccessful
+	else
+		res.jsonp({success: false});
+
+});
+
+//this method gets called before function body inside router.post, attemps to sign in user
+function signIn(req, res, next){
+	const email = req.body.email
+	const password = req.body.password;
+
+	firebase.auth().signInWithEmailAndPassword(email, password).catch(function(error) {
+		//sign in unsuccessful
+		console.log("invalid credential")
+	});
+
+	//set 1.2 second timeout so that above method can finish before moving on
+	setTimeout(next, 1200);
```

Comment:
`signInWithEmailAndPassword` is an [asynchronous function](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous) that returns a `Promise<UserCredential>`. If you're not clear how asynchronous functions (not to be confused with the `async` keyword) work, let me known so I can point out some resources because it's a pretty important concept in JavaScript. [Promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise), which `signInWithEmailAndPassword` and a lot of other libraries returns, builds on the idea of asynchronous functions, but basically, any code you want executed after `signInWithEmailAndPassword` is run should be in a `.then()` block (similar to the existing `.catch()` block, which is specifically for errors), so you probably want something more like
```javascript
function signIn(req, res, next) {
    const email = req.body.email;
    const password = req.body.password;
    firebase.auth().signInWithEmailAndPassword(email, password).then(function(firebaseUser) {
        next();
    }).catch(function(error) { ... });
}
```
where `next` runs after `signInWithEmailAndPassword` finishes (notice it's in the `.then()` block). That being said, `signIn` is a pretty small function and can just go in the route code.